src/System.Diagnostics.Process/tests/ProcessTests.cs

```diff
@@ -824,8 +824,7 @@ public void GetProcessesByName_ProcessNameMachineName_ReturnsExpected(string mac
             Assert.All(processes, process => Assert.Equal(machineName, process.MachineName));
         }
 
-        [ActiveIssue(18324)]
-        [Theory]
+        [ConditionalTheory(nameof(ProcessPeformanceCounterEnabled))]
```

Comment:
Or, you could keep the code as you'v ewritten it but replace `return !ex.Message.Contains("Process performance counter is disabled");` with `return false;`